### PR TITLE
feat(tradeshield): add extra info to perpetual pending orders and add…

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
… liq price to close estimation query

### What has Changed?

* add extra infos to pending perpetual orders
* add liquidation price to close estimation query in perpetual

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

**Review by amityadav0 — COMMENTED** [review]
I think we need to add a upgrade for perpetulOrder as new fields are added

**Comment by cosmic-vagabond** [comment]
> I think we need to add a upgrade for perpetulOrder as new fields are added

you are actually right we do need to add a migrator as the PerpetualOrder message is used within the module state, I will update the PR to include the missing migrator

File: proto/elys/tradeshield/types.proto [hunk 1]
```diff
@@ -85,4 +85,17 @@ message PerpetualOrder {
...
+        (gogoproto.nullable) = false
+    ];
+    string funding_rate = 15 [
+        (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+        (gogoproto.nullable) = false
+    ];
+    string borrow_interest_rate = 16 [
+        (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+        (gogoproto.nullable) = false
+    ];
```
Comment (amityadav0) [inline proto/elys/tradeshield/types.proto:100]: these fields

Comment (cosmic-vagabond) [reply proto/elys/tradeshield/types.proto:124]: @amityadav0 migrator was added here https://github.com/elys-network/elys/pull/847/commits/8665d7def3aebbe61a144c294e9e6dc3417c400f

**Review by amityadav0 — APPROVED** [review]
LGTM

**Pull request merged**